src/main/java/org/dependencytrack/tasks/repositories/NugetMetaAnalyzer.java

```diff
@@ -111,7 +111,10 @@ private boolean performVersionCheck(final MetaModel meta, final Component compon
                     String responseString = EntityUtils.toString(response.getEntity());
                     var jsonObject = new JSONObject(responseString);
                     final JSONArray versions = jsonObject.getJSONArray("versions");
-                    final String latest = findLatestVersion(versions); // get the last version in the array
+
+                    final boolean excludePreRelease = component.getPurl().getVersion() != null && component.getPurl().getVersion().contains("-") ==false;  // if the version is a pre-release version, we should not exclude pre-release versions
```

Comment:
Just a quick remark regarding Java standards. Rather than writing 
`component.getPurl().getVersion().contains("-") == false`
you should directly write 
`!component.getPurl().getVersion().contains("-")`
as it is already a boolean value.
Your code becomes 
`final boolean excludePreRelease = component.getPurl().getVersion() != null && !component.getPurl().getVersion().contains("-");` not much of a difference I admit, but it conforms better to java standards.

Reply:
Thanks @sebD, not worked in Java for a while so thanks for letting me know about the standards. I have refactored that code as per your suggestion.